common-build.gradle

```diff
@@ -47,3 +47,133 @@ task syncVersionAndroid  {
     }
 }
 
+task generateIOSBindings {
+    doLast() {
+        // Only generate bindings for iOS projects
+        if ( !project.hasProperty('iosSdkModuleName') ) {
+            return
+        }
+
+        println("Updating bindings for ${iosSdkModuleName} module.")
+
+        // clone iOS SDK at specified version, if it doesn't already exist
+        println("Using iOS SDK version ${airshipProperties.iosVersion}")
+        def tmpDir = System.getProperty("java.io.tmpdir")
+        def iosTempRepo = new File("${tmpDir}/ios-library-for-xamboni")
+        iosTempRepo.deleteOnExit()
+
+        boolean cloneRepo = false
+        if ( !iosTempRepo.exists() ) {
+            // cloned repo doesn't exist, so clone it
+            cloneRepo = true
+        } else {
+            // cloned repo exists, so check if it is the right version
+            def out = new ByteArrayOutputStream()
+            exec {
+                commandLine "git", "-C", "${iosTempRepo}", "tag", "--points-at", "HEAD"
+                standardOutput = out
+            }
+            def currentRepoHead = out.toString().replaceAll("\\s","")
+            if ( currentRepoHead == airshipProperties.iosVersion ) {
+                cloneRepo = false
+            } else {
+                iosTempRepo.deleteDir()
+                cloneRepo = true
+            }
+        }
+
+        // actually clone the repo, if necessary
+        if (cloneRepo) {
+            exec {
+                commandLine "git", "clone", "--branch", "${airshipProperties.iosVersion}", "https://github.com/urbanairship/ios-library", "${iosTempRepo}"
```

Comment:
is there any value in making this a separate task so that we have an easier time chaining it? I'm wondering what will happen if the clone step fails for some reason (network or otherwise). Would the gradle task exit cleanly if that were the case or would it result in a more obscure error further down?

Reply:
Done.